src/System.IO.FileSystem/tests/Directory/EnumerableTests.cs

```diff
@@ -19,7 +19,21 @@ public void FileEnumeratorIsThreadSafe()
             for (int i = 0; i < 100; i++)
                 File.Create(Path.Combine(directory, GetTestFileName())).Dispose();
 
-            new ThreadSafeRepro().Execute(directory);
+            // We are really only trying to make sure we don't terminate the process.
+            // Throwing IOException at this point to try and flush out other problems
+            // like bad handles. Can narrow the throw if this isn't reliable.
+
+            try
+            {
+                new ThreadSafeRepro().Execute(directory);
+            }
+            catch (IOException)
+            {
+                throw;
+            }
+            catch (Exception)
```

Comment:
This is fine.  Another approach would be:
```C#
catch (Exception e) when (!(e is IOException)) { }
```

Reply:
I'll update it as I'm updating VSB.